src/main/java/de/hpi/bpt/argos/api/eventQuery/EventQueryEndpointImpl.java

```diff
@@ -0,0 +1,198 @@
+        }
+        endpointUtil.logSendingResponse(logger, request, response.status(), response.body());
+        return "";
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public String deleteEventQuery(Request request, Response response) {
+        endpointUtil.logReceivedRequest(logger, request);
+
+        long eventQueryId = getEventQueryId(request);
+        EventQuery eventQuery = PersistenceAdapterImpl.getInstance().getEventQuery(eventQueryId);
+        if (eventQuery == null) {
+            halt(HttpStatusCodes.ERROR, "Given query was not found");
+        }
+
+        EventPlatformFeedback feedback = EventProcessingPlatformUpdaterImpl.getInstance().deleteEventQuery(eventQuery);
+        if (!feedback.isSuccessful()) {
+            halt(HttpStatusCodes.ERROR, String.format("cannot unregister event query: %1$s", feedback.getResponseText()));
+        }
+
+        if (!PersistenceAdapterImpl.getInstance().deleteArtifact(eventQuery,
+                EventTypeEndpoint.getEventTypeQueriesUri(eventQuery.getId()))) {
+            halt(HttpStatusCodes.ERROR, "Could not delete query");
+        }
+
+        endpointUtil.logSendingResponse(logger, request, response.status(), response.body());
+        return "";
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public String editEventQuery(Request request, Response response) {
+        endpointUtil.logReceivedRequest(logger, request);
+
+        JsonObject jsonBody = new JsonObject();
+        try {
+            jsonBody = jsonParser.parse(request.body()).getAsJsonObject();
+        } catch (JsonSyntaxException e) {
+            LoggerUtilImpl.getInstance().error(logger, "cannot parse request body to event query", e);
+            halt(HttpStatusCodes.BAD_REQUEST, "not a valid json");
+        }
+
+        JsonObject jsonEventQuery = jsonBody.get(JSON_EVENT_QUERY_ATTRIBUTE).getAsJsonObject();
+        if (jsonEventQuery == null) {
+            halt(HttpStatusCodes.BAD_REQUEST, "no event query given in body");
+        }
+
+        EventQuery newEventQuery = getEventQueryFromJson(jsonEventQuery);
+        if (newEventQuery == null) {
+            halt(HttpStatusCodes.BAD_REQUEST, "failed to parse event query");
+        }
+
+        long oldEventQueryId = getEventQueryId(request);
+        EventQuery oldEventQuery = PersistenceAdapterImpl.getInstance().getEventQuery(oldEventQueryId);
+        if (oldEventQuery == null) {
+            halt(HttpStatusCodes.ERROR, "Given query was not found");
+        }
+
+        EventPlatformFeedback deleteFeedback = EventProcessingPlatformUpdaterImpl.getInstance().deleteEventQuery(oldEventQuery);
+        if (!deleteFeedback.isSuccessful()) {
+            halt(HttpStatusCodes.ERROR, "Given query could not be unregistered");
+        }
+
+        EventPlatformFeedback createFeedback = EventProcessingPlatformUpdaterImpl.getInstance().registerEventQuery(newEventQuery.getTypeId(), newEventQuery);

```

Comment:
![BLOCKER](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png 'Severity: BLOCKER') NullPointerException might be thrown as 'newEventQuery' is nullable here [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://bpt-lab.org/sonarqube/coding_rules#rule_key=squid%3AS2259)
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Split this 157 characters long line (which is greater than 150 authorized). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://bpt-lab.org/sonarqube/coding_rules#rule_key=squid%3AS00103)